sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -93,7 +93,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
         RewriteCorrelatedScalarSubquery,
         EliminateSerialization,
         RemoveRedundantAliases,
-        RemoveRedundantProject,
+        RemoveNoopOperators,
```

Comment:
nit: `RemoveUselessOperators`?

Reply:
I think Noop is fine too. It's no-op, right :-)?

Reply:
yes it is fine too, I preferred `Useless` because they are actually doing something, so they introduce a useless overhead, anyway not a big deal